SignalMessaging/utils/ThreadUtil.m

```diff
@@ -93,44 +93,45 @@ + (TSOutgoingMessage *)enqueueMessageWithAttachment:(SignalAttachment *)attachme
                                            inThread:(TSThread *)thread
                                    quotedReplyModel:(nullable OWSQuotedReplyModel *)quotedReplyModel
 {
-    return [self enqueueMessageWithAttachment:attachment
-                                     inThread:thread
-                             quotedReplyModel:quotedReplyModel
-                                 ignoreErrors:NO];
+    return [self enqueueMessageWithAttachment:attachment inThread:thread quotedReplyModel:quotedReplyModel];
 }
 
-+ (TSOutgoingMessage *)enqueueMessageWithAttachment:(SignalAttachment *)attachment
-                                           inThread:(TSThread *)thread
-                                   quotedReplyModel:(nullable OWSQuotedReplyModel *)quotedReplyModel
-                                       ignoreErrors:(BOOL)ignoreErrors
++ (TSOutgoingMessage *)enqueueMessageWithAttachments:(NSArray<SignalAttachment *> *)attachments
+                                            inThread:(TSThread *)thread
+                                    quotedReplyModel:(nullable OWSQuotedReplyModel *)quotedReplyModel
 {
     OWSAssertIsOnMainThread();
-    OWSAssertDebug(attachment);
-    OWSAssertDebug(!attachment.hasError);
-    OWSAssertDebug(attachment.mimeType.length > 0);
+    OWSAssertDebug(attachments.count > 0);
     OWSAssertDebug(thread);
+    for (SignalAttachment *attachment in attachments) {
+        OWSAssertDebug(!attachment.hasError);
+        OWSAssertDebug(attachment.mimeType.length > 0);
+    }
 
     OWSDisappearingMessagesConfiguration *configuration =
         [OWSDisappearingMessagesConfiguration fetchObjectWithUniqueID:thread.uniqueId];
 
     uint32_t expiresInSeconds = (configuration.isEnabled ? configuration.durationSeconds : 0);
+    BOOL isVoiceMessage = (attachments.count == 1 && attachments.lastObject.isVoiceMessage);
+    // TODO: Support multi-image captions.
+    NSString *_Nullable messageBody = attachments.lastObject.captionText;
     TSOutgoingMessage *message =
         [[TSOutgoingMessage alloc] initOutgoingMessageWithTimestamp:[NSDate ows_millisecondTimeStamp]
                                                            inThread:thread
-                                                        messageBody:attachment.captionText
```

Comment:
So presumably the caption-text will eventually be a new field on the TSAttachment?

Reply:
It sounded like that might happen.